In [1]:
import pandas as pd

READING THE DATASET

In [2]:
df=pd.read_csv('merc.csv')

In [3]:
df

model  year  price transmission  mileage fuelType  tax   mpg  \
0             SLK  2005   5200    Automatic    63000   Petrol  325  32.1   
1         S Class  2017  34948    Automatic    27000   Hybrid   20  61.4   
2        SL CLASS  2016  49948    Automatic     6200   Petrol  555  28.0   
3         G Class  2016  61948    Automatic    16000   Petrol  325  30.4   
4         G Class  2016  73948    Automatic     4000   Petrol  325  30.1   
...           ...   ...    ...          ...      ...      ...  ...   ...   
13114     C Class  2020  35999    Automatic      500   Diesel  145  55.4   
13115     B Class  2020  24699    Automatic     2500   Diesel  145  55.4   
13116   GLC Class  2019  30999    Automatic    11612   Diesel  145  41.5   
13117   CLS Class  2019  37990    Automatic     2426   Diesel  145  45.6   
13118     S Class  2019  54999    Automatic     2075   Diesel  145  52.3   

       engineSize  
0             1.8  
1             2.1  
2             5.5  
3             4.0  
4             4.0  
...           ...  
13114         2.0  
13115         2.0  
13116         2.1  
13117         2.0  
13118         2.9  

[13119 rows x 9 columns]

UNDERSTANDING THE DATASET

In [4]:
df.columns

Index(['model', 'year', 'price', 'transmission', 'mileage', 'fuelType', 'tax',
       'mpg', 'engineSize'],
      dtype='object')

In [5]:
df.describe()

year          price        mileage           tax           mpg  \
count  13119.000000   13119.000000   13119.000000  13119.000000  13119.000000   
mean    2017.296288   24698.596920   21949.559037    129.972178     55.155843   
std        2.224709   11842.675542   21176.512267     65.260286     15.220082   
min     1970.000000     650.000000       1.000000      0.000000      1.100000   
25%     2016.000000   17450.000000    6097.500000    125.000000     45.600000   
50%     2018.000000   22480.000000   15189.000000    145.000000     56.500000   
75%     2019.000000   28980.000000   31779.500000    145.000000     64.200000   
max     2020.000000  159999.000000  259000.000000    580.000000    217.300000   

         engineSize  
count  13119.000000  
mean       2.071530  
std        0.572426  
min        0.000000  
25%        1.800000  
50%        2.000000  
75%        2.100000  
max        6.200000

HANDLING DUPLICATES AND MISSING VALUES

In [6]:
df.shape

(13119, 9)

In [8]:
duplicates = df[df.duplicated()]


In [9]:
duplicates

model  year  price transmission  mileage fuelType  tax   mpg  \
493       V Class  2018  19998       Manual    27222   Diesel  145  46.3   
494       V Class  2016  17998       Manual    21981   Diesel  200  46.3   
1328    GLB Class  2020  35400    Semi-Auto     2000   Diesel  145  46.3   
1505      E Class  2016  15995    Semi-Auto    60931   Diesel  125  58.9   
1506      C Class  2018  19950    Semi-Auto    33082   Diesel  145  64.2   
...           ...   ...    ...          ...      ...      ...  ...   ...   
10517     E Class  2015  18995    Automatic     7315   Diesel  160  50.4   
10518     E Class  2015  18995    Automatic     7315   Diesel  160  50.4   
11742     E Class  2016  17990    Automatic    39000   Diesel   30  65.7   
12800     B Class  2020  27490    Automatic     2563   Diesel  145  55.4   
12801     A Class  2020  26590    Automatic     2563   Petrol  145  53.3   

       engineSize  
493           2.1  
494           2.1  
1328          2.0  
1505          2.1  
1506          2.1  
...           ...  
10517         3.0  
10518         3.0  
11742         2.0  
12800         2.0  
12801         1.3  

[259 rows x 9 columns]

THERE ARE 259 DUPLICATE ENTRIES, SO WE NEED TO REMOVE THEM

In [10]:
df.drop_duplicates(inplace=True)

In [11]:
df.shape

(12860, 9)

In [12]:
df.isnull().sum()

model           0
year            0
price           0
transmission    0
mileage         0
fuelType        0
tax             0
mpg             0
engineSize      0
dtype: int64

THERE IS NO MISSING VALUES IN THE DATASET

In [13]:
df

model  year  price transmission  mileage fuelType  tax   mpg  \
0             SLK  2005   5200    Automatic    63000   Petrol  325  32.1   
1         S Class  2017  34948    Automatic    27000   Hybrid   20  61.4   
2        SL CLASS  2016  49948    Automatic     6200   Petrol  555  28.0   
3         G Class  2016  61948    Automatic    16000   Petrol  325  30.4   
4         G Class  2016  73948    Automatic     4000   Petrol  325  30.1   
...           ...   ...    ...          ...      ...      ...  ...   ...   
13114     C Class  2020  35999    Automatic      500   Diesel  145  55.4   
13115     B Class  2020  24699    Automatic     2500   Diesel  145  55.4   
13116   GLC Class  2019  30999    Automatic    11612   Diesel  145  41.5   
13117   CLS Class  2019  37990    Automatic     2426   Diesel  145  45.6   
13118     S Class  2019  54999    Automatic     2075   Diesel  145  52.3   

       engineSize  
0             1.8  
1             2.1  
2             5.5  
3             4.0  
4             4.0  
...           ...  
13114         2.0  
13115         2.0  
13116         2.1  
13117         2.0  
13118         2.9  

[12860 rows x 9 columns]

DON'T HAVE INFORMATION ON TAX COLUMN WHETHER IT IS IN % OR IN EUROS, SO DELETE IT

In [15]:
df.drop(['tax'],axis=1,inplace=True)

In [16]:
df

model  year  price transmission  mileage fuelType   mpg  \
0             SLK  2005   5200    Automatic    63000   Petrol  32.1   
1         S Class  2017  34948    Automatic    27000   Hybrid  61.4   
2        SL CLASS  2016  49948    Automatic     6200   Petrol  28.0   
3         G Class  2016  61948    Automatic    16000   Petrol  30.4   
4         G Class  2016  73948    Automatic     4000   Petrol  30.1   
...           ...   ...    ...          ...      ...      ...   ...   
13114     C Class  2020  35999    Automatic      500   Diesel  55.4   
13115     B Class  2020  24699    Automatic     2500   Diesel  55.4   
13116   GLC Class  2019  30999    Automatic    11612   Diesel  41.5   
13117   CLS Class  2019  37990    Automatic     2426   Diesel  45.6   
13118     S Class  2019  54999    Automatic     2075   Diesel  52.3   

       engineSize  
0             1.8  
1             2.1  
2             5.5  
3             4.0  
4             4.0  
...           ...  
13114         2.0  
13115         2.0  
13116         2.1  
13117         2.0  
13118         2.9  

[12860 rows x 8 columns]

In [18]:
df['model'].unique()

array([' SLK', ' S Class', ' SL CLASS', ' G Class', ' GLE Class',
       ' GLA Class', ' A Class', ' B Class', ' GLC Class', ' C Class',
       ' E Class', ' GL Class', ' CLS Class', ' CLC Class', ' CLA Class',
       ' V Class', ' M Class', ' CL Class', ' GLS Class', ' GLB Class',
       ' X-CLASS', '180', ' CLK', ' R Class', '230', '220', '200'],
      dtype=object)

In [17]:
df.to_csv('cleansed_merc.csv',index=False)